tests/explainers/test_tree.py

```diff
@@ -1197,6 +1197,30 @@ def test_xgboost_regression(self):
         expected_diff = np.abs(explanation.values.sum(1) + explanation.base_values - predicted).max()
         assert expected_diff < 1e-4, "SHAP values don't sum to model output!"
 
+    def test_xgboost_dmatrix_propagation(self):
+        xgboost = pytest.importorskip("xgboost")
+
+        def sigmoid(x):
+            return 1 / (1 + np.exp(-x))
+
+        X, y = shap.datasets.adult()
```

Comment:
Actually the `adult` dataset is relatively huge (32k rows) for a dataset to use in a unit test. If it's not critical, maybe we could bump the number of samples down a little, e.g. `shap.datasets.adult(n_samples=1000)` as long as it still reproduces the issue